[BUG] test_map_scalars_supported_key_types failed in dataproc 2.1

**Describe the bug**
related to https://github.com/NVIDIA/spark-rapids/pull/8971, the bug fix enabled cases `test_map_scalars_supported_key_types`

which failed in nightly dataproc pipeline `rapids-it-dataproc-2.1-ubuntu20` build: 130

```

[2023-08-13T13:24:02.349Z] =========================== short test summary info ============================

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Boolean(not_null),Boolean)][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Byte(not_null),Byte)] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Short(not_null),Short)][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Integer(not_null),Integer)] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Long(not_null),Long)] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Float(not_null),Float)] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Double(not_null),Double)][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Date(not_null),Date)] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Timestamp(not_null),Timestamp)] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(String(not_null),String)][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(String(not_null),Array(String))] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(RepeatSeq(not_null)(Integer(not_null)),Long)][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(String(not_null),Map(String(not_null),String))][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Integer(not_null),Array(Integer))][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Short(not_null),Struct(['child0', Byte],['child1', Double]))] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Byte(not_null),Map(Float(not_null),Date))][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(Integer(not_null),Binary)][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(not_null)(Decimal(not_null)(7,3),Decimal(not_null)(7,3))][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(not_null)(Decimal(not_null)(12,2),Decimal(not_null)(12,2))][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] FAILED ../../src/main/python/map_test.py::test_map_scalars_supported_key_types[Map(not_null)(Decimal(not_null)(20,2),Decimal(not_null)(20,2))][INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columna...

[2023-08-13T13:24:02.349Z] = 20 failed, 513 passed, 5 skipped, 20248 deselected, 9 warnings in 614.87s (0:10:14) =
```

```

[2023-08-13T13:24:02.348Z] ----------------------------- Captured stdout call -----------------------------

[2023-08-13T13:24:02.348Z] ### CPU RUN ###

[2023-08-13T13:24:02.348Z] ### GPU RUN ###

[2023-08-13T13:24:02.348Z] _ test_map_scalars_supported_key_types[Map(not_null)(Decimal(not_null)(12,2),Decimal(not_null)(12,2))] _

[2023-08-13T13:24:02.348Z] 

[2023-08-13T13:24:02.348Z] data_gen = Map(not_null)(Decimal(not_null)(12,2),Decimal(not_null)(12,2))

[2023-08-13T13:24:02.348Z] 

[2023-08-13T13:24:02.348Z]     @pytest.mark.parametrize('data_gen', supported_key_map_gens, ids=idfn)

[2023-08-13T13:24:02.348Z]     def test_map_scalars_supported_key_types(data_gen):

[2023-08-13T13:24:02.348Z]         key_gen = data_gen._key_gen

[2023-08-13T13:24:02.348Z]         def query_map_scalar(spark):

[2023-08-13T13:24:02.348Z]             key_df = gen_df(spark, [("key", key_gen)], length=100).orderBy(col("key"))\

[2023-08-13T13:24:02.348Z]                 .select(col("key"),

[2023-08-13T13:24:02.348Z]                         row_number().over(Window().orderBy(col('key')))\

[2023-08-13T13:24:02.348Z]                             .alias("row_num"))

[2023-08-13T13:24:02.348Z]             key_df.select(col("key").alias("key_at_ix"))\

[2023-08-13T13:24:02.348Z]                 .where(col("row_num") == 5)\

[2023-08-13T13:24:02.348Z]                 .repartition(100)\

[2023-08-13T13:24:02.348Z]                 .createOrReplaceTempView("single_key_tbl")

[2023-08-13T13:24:02.348Z]             key_df.select(col("key").alias("key_at_ix_next")) \

[2023-08-13T13:24:02.348Z]                 .where(col("row_num") == 6) \

[2023-08-13T13:24:02.348Z]                 .repartition(100) \

[2023-08-13T13:24:02.348Z]                 .createOrReplaceTempView("single_key_tbl_next")

[2023-08-13T13:24:02.348Z]             # There will be a single row in single_key_tbl (the row that matched key_ix).

[2023-08-13T13:24:02.348Z]             # We repartition this table to create several empty tables to test empty partitions,

[2023-08-13T13:24:02.348Z]             # and also because the window operation put everything into a one partition prior.

[2023-08-13T13:24:02.348Z]             # Because this is a single key, first(ignore_nulls = true) will be deterministic.

[2023-08-13T13:24:02.348Z]             return spark.sql(

[2023-08-13T13:24:02.348Z]                 "select key_at_ix, " +

[2023-08-13T13:24:02.348Z]                 "       (select first(map(key_at_ix, 'value'), true) " +

[2023-08-13T13:24:02.348Z]                 "        from single_key_tbl)[key_at_ix], " +

[2023-08-13T13:24:02.348Z]                 # this one is on purpose using `key_at_ix` to guarantee we won't match the key

[2023-08-13T13:24:02.348Z]                 "       (select first(map(key_at_ix_next, 'value'), true) " +

[2023-08-13T13:24:02.348Z]                 "        from single_key_tbl_next)[key_at_ix] " +

[2023-08-13T13:24:02.348Z]                 "from single_key_tbl")

[2023-08-13T13:24:02.348Z] >       assert_cpu_and_gpu_are_equal_collect_with_capture(

[2023-08-13T13:24:02.348Z]             query_map_scalar,

[2023-08-13T13:24:02.348Z]             # check that GpuGetMapValue wasn't optimized out

[2023-08-13T13:24:02.348Z]             exist_classes="GpuGetMapValue",

[2023-08-13T13:24:02.348Z]             conf = {"spark.rapids.sql.explain": "NONE"})

[2023-08-13T13:24:02.348Z] 

[2023-08-13T13:24:02.348Z] ../../src/main/python/map_test.py:216: 

[2023-08-13T13:24:02.348Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2023-08-13T13:24:02.348Z] ../../src/main/python/asserts.py:382: in assert_cpu_and_gpu_are_equal_collect_with_capture

[2023-08-13T13:24:02.348Z]     from_gpu, gpu_df = with_gpu_session(bring_back, conf=conf)

[2023-08-13T13:24:02.348Z] ../../src/main/python/spark_session.py:133: in with_gpu_session

[2023-08-13T13:24:02.348Z]     return with_spark_session(func, conf=copy)

[2023-08-13T13:24:02.348Z] ../../src/main/python/spark_session.py:100: in with_spark_session

[2023-08-13T13:24:02.348Z]     ret = func(_spark)

[2023-08-13T13:24:02.348Z] ../../src/main/python/asserts.py:208: in bring_back

[2023-08-13T13:24:02.348Z]     return (df.collect(), df)

[2023-08-13T13:24:02.348Z] /usr/lib/spark/python/lib/pyspark.zip/pyspark/sql/dataframe.py:817: in collect

[2023-08-13T13:24:02.348Z]     sock_info = self._jdf.collectToPython()

[2023-08-13T13:24:02.348Z] /usr/lib/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py:1321: in __call__

[2023-08-13T13:24:02.348Z]     return_value = get_return_value(

[2023-08-13T13:24:02.348Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2023-08-13T13:24:02.348Z] 

[2023-08-13T13:24:02.348Z] a = ('xro63325', <py4j.clientserver.JavaClient object at 0x7f0e866f6380>, 'o63324', 'collectToPython')

[2023-08-13T13:24:02.348Z] kw = {}, converted = IllegalArgumentException()

[2023-08-13T13:24:02.348Z] 

[2023-08-13T13:24:02.348Z]     def deco(*a: Any, **kw: Any) -> Any:

[2023-08-13T13:24:02.348Z]         try:

[2023-08-13T13:24:02.348Z]             return f(*a, **kw)

[2023-08-13T13:24:02.348Z]         except Py4JJavaError as e:

[2023-08-13T13:24:02.348Z]             converted = convert_exception(e.java_exception)

[2023-08-13T13:24:02.348Z]             if not isinstance(converted, UnknownException):

[2023-08-13T13:24:02.348Z]                 # Hide where the exception came from that shows a non-Pythonic

[2023-08-13T13:24:02.348Z]                 # JVM exception message.

[2023-08-13T13:24:02.348Z] >               raise converted from None

[2023-08-13T13:24:02.348Z] E               pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.window.WindowLocalExec

[2023-08-13T13:24:02.348Z] E               WindowLocal [row_number() windowspecdefinition(key#6471 ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS row_num#6458], [key#6471 ASC NULLS FIRST]

[2023-08-13T13:24:02.348Z] E               +- GpuColumnarToRow false

[2023-08-13T13:24:02.348Z] E                  +- GpuSort [key#6471 ASC NULLS FIRST], false, com.nvidia.spark.rapids.OutOfCoreSort$@57362536

[2023-08-13T13:24:02.348Z] E                     +- GpuSort [key#6471 ASC NULLS FIRST], true, com.nvidia.spark.rapids.OutOfCoreSort$@57362536

[2023-08-13T13:24:02.348Z] E                        +- GpuShuffleCoalesce 104857600

[2023-08-13T13:24:02.348Z] E                           +- GpuColumnarExchange gpurangepartitioning(key#6471 ASC NULLS FIRST, 4), ENSURE_REQUIREMENTS, [id=#27175]

[2023-08-13T13:24:02.348Z] E                              +- GpuRowToColumnar targetsize(104857600)

[2023-08-13T13:24:02.348Z] E                                 +- Scan ExistingRDD[key#6471]

[2023-08-13T13:24:02.348Z] 

[2023-08-13T13:24:02.348Z] /usr/lib/spark/python/lib/pyspark.zip/pyspark/sql/utils.py:196: IllegalArgumentException
```

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

**Expected behavior**
Pass the test

**Environment details (please complete the following information)**
Dataproc

**Additional context**
Add any other context about the problem here.

--- a/integration_tests/src/main/python/map_test.py
+++ b/integration_tests/src/main/python/map_test.py
@@ -185,6 +185,7 @@ def test_basic_scalar_map_get_map_value(key_gen):
                 "spark.rapids.sql.castFloatToIntegralTypes.enabled": True})
 
 
+@allow_non_gpu('WindowLocalExec')
 @pytest.mark.parametrize('data_gen', supported_key_map_gens, ids=idfn)
 def test_map_scalars_supported_key_types(data_gen):
     key_gen = data_gen._key_gen

